core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -731,9 +731,9 @@ public void setColumns(List<String> columnNames, List<Type> columnTypes)
         outputManager.setColumns(columnNames, columnTypes);
     }
 
-    public void updateOutputLocations(Map<TaskId, URI> newExchangeLocations, boolean noMoreExchangeLocations)
+    public void updateOutputLocations(Set<ExchangeInput> inputs, boolean noMoreInputs)
```

Comment:
should this one be renamed now?

Reply:
tricky.... maybe `updateInputsForQueryOutput`. Or maybe `updateInputsForQueryResults` - or would that be confusing?

Reply:
`updateInputsForQueryResults` sounds great, let me do the rename